inference/huggingface/text-generation/inference-test.py

```diff
@@ -78,21 +95,41 @@
 
 inputs = input_sentences[:args.batch_size]
 
-# warmup
-outputs = pipe(inputs,
-              num_tokens=args.max_new_tokens,
-              do_sample=(not args.greedy))
+if args.test_performance:
+    times = []
+    for i in range(30):
+        torch.cuda.synchronize()
+        start = time.time()
+
+        outputs = pipe(inputs,
+                num_tokens=args.max_new_tokens,
+                do_sample=(not args.greedy))
+        
+        torch.cuda.synchronize()
+        end = time.time()
+
+        times.append(end - start)
+
+    if args.local_rank == 0:
+        print_perf_stats(map(lambda t: t / (args.max_new_tokens),
+                    times),
+                    pipe.model.config)
+else:
+    # warmup
+    outputs = pipe(inputs,
+                num_tokens=args.max_new_tokens,
+                do_sample=(not args.greedy))
 
-torch.cuda.synchronize()
-start = time.time()
+    torch.cuda.synchronize()
+    start = time.time()
 
-outputs = pipe(inputs,
-              num_tokens=args.max_new_tokens,
-              do_sample=(not args.greedy))
+    outputs = pipe(inputs,
+                num_tokens=args.max_new_tokens,
+                do_sample=(not args.greedy))
 
-torch.cuda.synchronize()
-end = time.time()
-print(f'generation time is {end-start} sec')
+    torch.cuda.synchronize()
+    end = time.time()
+    print(f'generation time is {end-start} sec')
```

Comment:
I've run into problems in the past where we have this copied code structure in an `if...else` statement and I may update one but forget to update the other (e.g., the call to `pipe`). We could simplify this block of code and avoid that potential problem in the future with:
```python
iters = 30 if args.test_performance else 1
times = []
for i in range(iters):
    torch.cuda.synchronize()
    start = time.time()
    outputs = pipe(inputs,
            num_tokens=args.max_new_tokens,
            do_sample=(not args.greedy))
    torch.cuda.synchronize()
    end = time.time()
    times.append(end - start)

if args.local_rank == 0:
    if args.test_performance:
        print_perf_stats(map(lambda t: t / args.max_new_tokens, times), pipe.model.config)
    else:
        print(f"generation time is {times[0]} sec")
```